src/rs_quark_assets/components/A2HSBanner.vue

```diff
@@ -0,0 +1,151 @@
+          aria-live="assertive"
+          class="ios-center max-w-sm overflow-hidden rounded-lg bg-white shadow-lg ring-1 ring-black ring-opacity-5"
+        >
+          <div class="p-4">
+            <div class="flex items-center gap-4">
+              <div class="rounded-full border-2 border-green-400 p-2">
+                <SvgCheck class="cursor-pointer text-green-400" />
+              </div>
+              <div>
+                <p class="text-lg font-medium text-muted">
+                  {{ messages.ios }}
+                </p>
+              </div>
+              <button
+                class="inline-flex"
+                @click="emit('triggered', 'triggered')"
+              >
+                <span class="sr-only">Close</span>
+                <SvgX
+                  class="cursor-pointer text-blue-500 hover:text-blue-700"
+                />
+              </button>
+            </div>
+          </div>
+        </div>
+        <div
+          v-else-if="unsupportedIos"
+          aria-live="assertive"
+          class="max-w-sm overflow-hidden rounded-lg bg-white shadow-lg ring-1 ring-black ring-opacity-5"
+        >
+          <div class="p-4">
+            <div class="flex items-center gap-4">
+              <div class="alert--error rounded-full border-2 p-2">
+                <SvgHandStop class="alert--error cursor-pointer" />
+              </div>
+              <div>
+                <p class="text-lg font-medium text-muted">
+                  {{ messages.iosUnsupported }}
+                </p>
+              </div>
+              <button class="inline-flex" @click="emit('triggered')">
+                <span class="sr-only">Close</span>
+                <SvgX
+                  class="cursor-pointer text-blue-500 hover:text-blue-700"
+                />
+              </button>
+            </div>
+          </div>
+        </div>
+      </div>
+    </transition>
+  </div>
+</template>
+
+<script>
+export default {
+  name: "DLA2HS",
+}
+</script>
+
+<script setup>
+import { ref } from "vue"
+
+import SvgCheck from "../assets/images/check.svg"
+import SvgX from "../assets/images/x.svg"
+import Bowser from "bowser"
+
+import SvgHandStop from "../assets/images/hand-stop.svg"
+
+const props = defineProps({
+  installPromptEvent: { type: Object, required: true },
+  eventTriggered: { type: String, required: true },
+})
+const emit = defineEmits(["triggered", "installPrompt"])
+
+const show = ref(false)
+const browser = Bowser.getParser(window.navigator.userAgent)
+
+const isIosSafari = browser.satisfies({
+  ios: {
+    safari: ">=11.3",
+  },
+})
+
+const unsupportedIos =
+  browser.getOS().name === "iOS" && browser.getBrowser.name !== "Safari"
+const messages = {
+  installPrompt: "Install Quark on your device.",
+  ios: "Install Quark on your device: tap Share and then Add to Home Screen",
+  iosUnsupported: "Use Safari to install this app to the home screen.",
+}
+
+const useA2hs = () => {
```

Comment:
```suggestion
function addToHomescreen() {
```

the `use` prefix is an indicator for composables, this is just a Function